src/variables/components/UpdateVariableOverlay.tsx

```diff
@@ -129,30 +129,31 @@ class UpdateVariableOverlay extends PureComponent<Props, State> {
                       />
                     </Grid.Column>
                   </Grid.Row>
-                  <Grid.Row>
-                    <Grid.Column>
-                      <Form.Footer>
-                        <Button
-                          text="Cancel"
-                          color={ComponentColor.Danger}
-                          onClick={onClose}
-                        />
-                        <Button
-                          text="Submit"
-                          type={ButtonType.Submit}
-                          color={ComponentColor.Primary}
-                          status={
-                            hasValidArgs
-                              ? ComponentStatus.Default
-                              : ComponentStatus.Disabled
-                          }
-                        />
-                      </Form.Footer>
-                    </Grid.Column>
-                  </Grid.Row>
                 </Grid>
               </Form>
             </Overlay.Body>
+            <Overlay.Footer>
+              <Button
+                text="Cancel"
+                color={ComponentColor.Secondary}
+                onClick={onClose}
+              />
+              <Button
+                text="Submit"
+                type={ButtonType.Submit}
+                color={ComponentColor.Primary}
+                status={
+                  hasValidArgs
+                    ? ComponentStatus.Default
+                    : ComponentStatus.Disabled
+                }
+                onClick={e => {
+                  this.handleSubmit(e)
+                  onClose()
```

Comment:
let's pull this out into a function that isn't re-created on render. This will cause re-renders since an anonymous function is being re-created on every render, which will be a different prop for `Button` on ever render.

Reply:
We will have to leave this one because of how the react context works, the `onClose` is not accessible outside of the arrow function. The other two instances of where re-renders could be avoided, have been fixed 👍